### PR TITLE
Add spec to compare answers/config to migrations

### PR DESCRIPTION
**Pull request by stbenjam**
This ensures answers/config are up-to-date, and also effectively
performs a test that the migrations themselves are valid.

### CODE REVIEW

**Comment by stbenjam** [comment]
The `rake config:migrate` task that applies the migrations, so you don't have to do it manually, wipes out comments unfortunately.  This wasn't a problem in katello-installer, as we don't have any.

But the tests themselves are maybe useful to foreman-installer?

File: Rakefile [hunk 1]
```diff
@@ -167,3 +177,23 @@ namespace :pkg do
     `bzip2 -9 #{PKGDIR}/foreman-installer-#{version}.tar`
   end
 end
+
+
+namespace :config do
+  task :migrate do
```
Comment (domcleal) [inline Rakefile]: Isn't this the same as `bin/foreman-installer --migrations-only`?

(I'm hesitant to add code using Kafo internals if unnecessary)

**Review by domcleal — CHANGES_REQUESTED** [review]

File: Rakefile (hunk 1 above)
Comment (stbenjam) [reply Rakefile]: Yes, for foreman-installer it is.  Removed it.

File: Rakefile [hunk 2]
```diff
@@ -1,5 +1,12 @@
 require 'rake/clean'
 require 'yaml'
+require 'kafo'
```
Comment (domcleal) [inline Rakefile]: Unnecessary now the task using it's been removed.

**Review by domcleal — CHANGES_REQUESTED** [review]

**Review by mmoll — APPROVED** [review]

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Thanks @stbenjam. Please remember to reply to PR comments so we know to re-review.

**Comment by stbenjam** [comment]
Label changes trigger a notification, don't they? Maybe not.. will make sure to do that then.

**Comment by domcleal** [comment]
No, they don't.

**Comment by stbenjam** [comment]
I don't rely on github notifications, so I didn't realize labels don't trigger a notification. What's the point of labeling something "Needs re-review" if it doesn't trigger any notification and no one looks at the label? Seems like a problem with theforeman-bot, not me.

I use bugwarrior/taskwarrior so I do indeed notice when PR's I'm involved in switch labels and require an action from me.

**Pull request merged**